pkg/util/limiter/response_size_limiter.go

```diff
@@ -0,0 +1,54 @@
+package limiter
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/atomic"
+)
+
+type responseSizeLimiterCtxKey struct{}
+
+var (
+	responseLimiterCtxKey = &responseSizeLimiterCtxKey{}
+	ErrMaxResponseSizeHit = "the query response size exceeds limit (limit: %d bytes)"
+)
+
+type ResponseSizeLimiter struct {
+	responseSizeCount atomic.Int64
+	maxResponseSize   int
```

Comment:
Can you use int64 instead of just int?
It is unlikely to happen but for those 32-bit arch maxResponseSize can overflow > 2GB.